docs/user/migration/external-name-alias.md

```diff
@@ -0,0 +1,154 @@
+# Migration guide for external name alias behavior change
+
+## Context
+
+Istio with 1.21 version has changed the behavior of the Service of type ExternalName.
+Previously, for the Service of type ExternalName a separate `cluster` in the Envoy config was created.
+With the change, the Service of type ExternalName is treated as an alias of the Service that it points to.
+This change has been introduced to align the behavior of Istio with handling of the Service of type ExternalName with the Kubernetes behavior.
+Since it caused some issues for our users, we introduced a new annotation "disable-external-name-alias" to disable this behavior.
+Under the hood, we apply `ENABLE_EXTERNAL_NAME_ALIAS` environment variable on the Istiod component.
+At certain point Istio will drop the support for this environment variable, making the behavior change obligatory.
+Currently, it's already removed from the main branch with the [PR](https://github.com/istio/istio/pull/52317).
```

Comment:
I would not mention that in this customer facing documentation, the state of the PR might change, and anyway it needs to have a release and we need to bump that

Reply:
This PR is already merged.

Reply:
We can remove this from this tutorial, I put it here to bring more context and insight into what's the situation. Unfortunately it is already dropped and I doubt if it will be back again.